diff --git a/src/objnam.c b/src/objnam.c
index f9bc1749f..97b779bbc 100644
--- a/src/objnam.c
+++ b/src/objnam.c
@@ -2,20 +2,24 @@
 /* Copyright (c) Stichting Mathematisch Centrum, Amsterdam, 1985. */
 /*-Copyright (c) Robert Patrick Rankin, 2011. */
 /* NetHack may be freely redistributed.  See license for details. */
 
 #include "hack.h"
 
 /* "an uncursed greased partly eaten guardian naga hatchling [corpse]" */
 #define PREFIX 80 /* (56) */
 #define SCHAR_LIM 127
 #define NUMOBUF 12
+#undef OBJ_NAME
+#undef OBJ_DESCR
+#define OBJ_NAME(obj) (tra("object_name", obj_descr[(obj).oc_name_idx].oc_name))
+#define OBJ_DESCR(obj) (tra("object_name", obj_descr[(obj).oc_descr_idx].oc_descr))
 
 struct _readobjnam_data {
     struct obj *otmp;
     char *bp;
     char *origbp;
     char oclass;
     char *un, *dn, *actualn;
     const char *name;
     char *p;
     int cnt, spe, spesgn, typ, very, rechrg;
@@ -2828,20 +2832,21 @@ singplur_compound(char *str)
  * genocidable, and vanquished-mon handling does its own special casing
  * (for uniques who've been revived and re-killed), so we don't bother
  * trying to get those right here.
  *
  * Also misused by muse.c to convert 1st person present verbs to 2nd person.
  * 3.6.0: made case-insensitive.
  */
 char *
 makeplural(const char *oldstr)
 {
+    return oldstr;
     char *spot;
     char lo_c, *str = nextobuf();
     const char *excess = (char *) 0;
     int len, i;
 
     if (oldstr)
         while (*oldstr == ' ')
             oldstr++;
     if (!oldstr || !*oldstr) {
         impossible("plural of null?");
@@ -5421,21 +5426,21 @@ rnd_class(int first, int last)
 const char *
 Japanese_item_name(int i, const char *ordinaryname)
 {
     const struct Jitem *j = Japanese_items;
 
     while (j->item) {
         if (i == j->item)
             return j->name;
         j++;
     }
-    return ordinaryname;
+    return tra("object_japanese_name", ordinaryname);
 }
 
 const char *
 armor_simple_name(struct obj *armor)
 {
     const char *result = 0;
     unsigned armcat = objects[armor->otyp].oc_armcat;
 
     switch (armcat) {
     case ARM_SUIT:
