diff --git a/src/do_name.c b/src/do_name.c
index 75f33815f..2d9872d7b 100644
--- a/src/do_name.c
+++ b/src/do_name.c
@@ -1297,21 +1297,21 @@ Mgender(struct monst *mtmp)
         mgender = FEMALE;
     }
     return mgender;
 }
 
 const char *
 pmname(struct permonst *pm, int mgender)
 {
     if (mgender < MALE || mgender >= NUM_MGENDERS || !pm->pmnames[mgender])
         mgender = NEUTRAL;
-    return pm->pmnames[mgender];
+    return tra("pm_name", pm->pmnames[mgender]);
 }
 #endif /* PMNAME_MACROS */
 
 /* mons[]->pmname for a monster */
 const char *
 mon_pmname(struct monst *mon)
 {
     /* for neuter, mon->data->pmnames[MALE] will be Null and use [NEUTRAL] */
     return pmname(mon->data, Mgender(mon));
 }
